### PR TITLE
scheduler: do not remove the operator when the step does not finish

### PR DESCRIPTION
**Pull request by shafreeck**
Signed-off-by: Shafreeck Sea <shafreeck@gmail.com>

### What problem does this PR solve? 

There is a bug introduced by #1652, in some case, like adding peers or adding learners, the step is left unfinished if the peer is in pending state, although the conf version has changed, in these cases, the  operator will be removed because the controller thought someone has  changed the conf version(in fact, it self did). We fix that by checking if the conf version has actually changed by current step, if it is, the operator is not regarded as stale.

### What is changed and how it works?

Only check conf version changes when a new step to be executed.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/schedule/operator_controller.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func (oc *OperatorController) Dispatch(region *core.RegionInfo, source string) {
 	// Check existed operator.
 	if op := oc.GetOperator(region.GetID()); op != nil {
 		timeout := op.IsTimeout()
-		if step := op.Check(region); step != nil && !timeout {
+		if advanced, step := op.Check(region); step != nil && !timeout {
```
Comment (shafreeck) [inline server/schedule/operator_controller.go]: There may be a problem here if the current step does not finish, should we send it again?

**Comment by nolouch** [comment]
@Luffbee Only if there are two operator works on one region at the same time. but consider the network, It's possible but the probability is very small. But the`epoch` of the region will monotonically increase, so as your suggestion may better than now.

**Comment by shafreeck** [comment]
@Luffbee I thought about your suggestion before and gave up. From the perspective of code, they really can be merged, but it is weird according to the semantics.  Conf version is not the completion of an epoch. It is acceptable to return if the epoched changed like 

`func IsFinish(region) (epochChanged bool, finished bool)`

It is weird like this

 `func IsFinish(region) (confVerChanged bool, finished bool)`

**Comment by nolouch** [comment]
PTAL @Luffbee @rleungx

**Pull request merged**